app/src/main/java/org/mozilla/fenix/SearchWidgetProvider.kt

```diff
@@ -80,25 +87,32 @@ class SearchWidgetProvider : AppWidgetProvider() {
             }
     }
 
-    private fun createVoiceSearchIntent(context: Context): PendingIntent {
-        return Intent(context, IntentReceiverActivity::class.java)
+    private fun createVoiceSearchIntent(context: Context): PendingIntent? {
+        val voiceIntent = Intent(context, IntentReceiverActivity::class.java)
             .let { intent ->
                 intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
                 intent.putExtra(IntentReceiverActivity.SPEECH_PROCESSING, true)
-                PendingIntent.getActivity(context, REQUEST_CODE_VOICE, intent, 0)
             }
+
+        val intentSpeech = Intent(RecognizerIntent.ACTION_RECOGNIZE_SPEECH)
+        if (intentSpeech.resolveActivity(context.packageManager) == null) {
```

Comment:
A convenient shortcut here would be to use `let` with a safe call to return null if the first expression is null.
```
return intentSpeech.resolveActivity(context.packageManager)?.let {
   PendingIntent.getActivity(context, REQUEST_CODE_VOICE, voiceIntent, 0)
}
```